File: ball.py
import random

class Ball:
    def __init__(self, canvas):
        self.canvas = canvas
        self.reset()
        self.shape = canvas.create_oval(
            self.x - self.radius,
            self.y - self.radius,
            self.x + self.radius,
            self.y + self.radius,
            fill="white"
        )
    
    def reset(self):
        self.x = 300
        self.y = 200
        self.radius = 10
        self.speed = 5
        self.dx = self.speed * random.choice([-1, 1])
        self.dy = self.speed * random.choice([-1, 1])
    
    def move(self):
        self.x += self.dx
        self.y += self.dy
        
        # Bounce off top and bottom
        if self.y <= self.radius or self.y >= 400 - self.radius:
            self.dy *= -1
        
        self.canvas.coords(
            self.shape,
            self.x - self.radius,
            self.y - self.radius,
            self.x + self.radius,
            self.y + self.radius
        )
    
    def collides_with(self, paddle):
        paddle_coords = self.canvas.coords(paddle.shape)
        ball_coords = self.canvas.coords(self.shape)
        
        return (
            ball_coords[2] >= paddle_coords[0] and
            ball_coords[0] <= paddle_coords[2] and
            ball_coords[3] >= paddle_coords[1] and
            ball_coords[1] <= paddle_coords[3]
        )

File: js/ball.js
export class Ball {
    constructor(config) {
        this.config = config;
        this.reset();
    }

    reset() {
        this.x = this.config.CANVAS_WIDTH / 2;
        this.y = this.config.CANVAS_HEIGHT / 2;
        this.radius = this.config.BALL_RADIUS;
        this.dx = this.config.BALL_SPEED * (Math.random() > 0.5 ? 1 : -1);
        this.dy = this.config.BALL_SPEED * (Math.random() > 0.5 ? 1 : -1);
    }

    draw(ctx) {
        ctx.beginPath();
        ctx.arc(this.x, this.y, this.radius, 0, Math.PI * 2);
        ctx.fillStyle = 'white';
        ctx.fill();
        ctx.closePath();
    }

    move() {
        this.x += this.dx;
        this.y += this.dy;

        if (this.y - this.radius < 0 || this.y + this.radius > this.config.CANVAS_HEIGHT) {
            this.dy = -this.dy;
        }
    }

    collidesWith(paddle) {
        return this.x + this.radius > paddle.x &&
               this.x - this.radius < paddle.x + paddle.width &&
               this.y + this.radius > paddle.y &&
               this.y - this.radius < paddle.y + paddle.height;
    }
}